### DISCUSSION TITLE
How to create quarterly partitions

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17630] [Q&A] [answered]
If I wanted an asset partitioned quarterly that worked the same as `MonthlyPartitionsDefinition`, would it be best to write a `QuarterlyPartitionsDefinition` setup or is there a way to use one of the existing partition definition items to set "arbitrary" date ranges to get quarterly partitions? We have a huge amount of quarterly reporting, which is why I'm asking.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1698871988643619?thread_ts=1698871988.643619&cid=C01U954MEER).*

**Comment by sryza** [comment, marked as answer]
You should be able to make a quarterly partitions definition with

```python
from dagster import TimeWindowPartitionsDefinition

partitions_def = TimeWindowPartitionsDefinition(cron_schedule="0 0 1 */3 *", start="2023-01", fmt="%Y-%m")

```

Reference: https://crontab.guru/every-quarter

**Reply by clayheaton** [reply]
Thank you! I did not see `TimeWindowPartitionsDefinition` on the documentation page, which is what left me wondering. 

https://docs.dagster.io/concepts/partitions-schedules-sensors/partitioning-assets

Here's the API docs page for it:

https://docs.dagster.io/_apidocs/partitions#dagster.TimeWindowPartitionsDefinition
